types/conf_test.go

```diff
@@ -108,6 +199,33 @@ var _ = Describe("config operations", func() {
 		Expect(err).To(HaveOccurred())
 	})
 
+	It("fails when delegate field exists but fields are named incorrectly", func() {
+		conf := `{
+	"name": "node-cni-network",
+		"type": "multus",
+    "kubeconfig": "/etc/kubernetes/node-kubeconfig.yaml",
+		"prevResult": {
+			"ips": [
+				{
+					"version": "4",
+					"address": "10.0.0.5/32",
+					"interface": 2
+				}
+			]
+		},
+		"delegates": [{
+			"thejohn": "weave-net"
```

Comment:
nit: change to `"_not_type"` form `"thejohn"` or something so that we know it's essentially an invalid or missing `"type"` field.